src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax;
+
+public class FigureSupplier {
+    public static final Color WHITE = Color.WHITE;
+    public static final int DEFAULT_FIGURE_RADIUS = 10;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int type = (int) (Math.random() * 5);
+        Color color = colorSupplier.getRandomColor();
+
+        switch (type) {
+            case 0:
+                double side = Math.random() * 10 + 1;
+                return new Square(color, side);
```

Comment:
Move each action to the private method
```suggestion
                return prepareSquare(color);
```
All private methods should be below public methods